odl/deform/optimal_information_transport.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Shape-based reconstruction using optimal information transportation.
+
+The Fisher-Rao metric is used in regularization term. And L2 data matching
+term is used in fitting term.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+import numpy as np
+from odl.discr import Gradient
+from odl.trafos import FourierTransform
+import odl
+standard_library.install_aliases()
+
+__all__ = ('optimal_information_transport_solver',)
+
+
+def optimal_information_transport_solver(gradS, I, niter, eps,
+                                         sigma, callback=None):
+    """
```

Comment:
Please consult the [How to document](http://odl.readthedocs.io/dev/document.html) guide for information on how to write function docstrings.